internal/k8shandler/deployment.go

```diff
@@ -357,7 +357,15 @@ func (node *deploymentNode) executeUpdate() error {
 		}
 
 		if ArePodTemplateSpecDifferent(currentDeployment.Spec.Template, node.self.Spec.Template) {
+			volumes := currentDeployment.Spec.Template.Spec.Volumes
+			hasVolumes := containsUsedVolumeStorage(currentDeployment.Spec.Template)
+
 			currentDeployment.Spec.Template = node.self.Spec.Template
+
+			if hasVolumes {
+				currentDeployment.Spec.Template.Spec.Volumes = volumes
+			}
+
```

Comment:
could we instead push this logic into `ArePodTemplateSpecDifferent` since it looks like we're repeating this logic both here and in the statefulset file

Reply:
I'll make a new function like `CopyOverVolume` or something like that in that file to reduce the cookie-cutter code. My concern moving it into that function is that if something outside the the volumes needs to change, then the volumes would still get overwritten (to my understanding).